src/Common/MemoryTracker.cpp

```diff
@@ -187,7 +187,7 @@ void MemoryTracker::free(Int64 size)
         DB::TraceCollector::collect(DB::TraceType::MemorySample, StackTrace(), -size);
     }
 
-    if (level == VariableContext::Thread)
+    if (parent_level == VariableContext::Thread)
```

Comment:
Don't understand how initial level can be anything other than thread.
I assume that memory tracking always goes via `DB::CurrentThread::getMemoryTracker()`

It means we don't need this branch anymore...

Reply:
Hm, nice catch, Ok, this PR will be replaced with these two patches from #16121:
- 17dcdaef751e2eaa60e6f5ed63779dfdafc81eaf
- 52d247f828c2c297dd4e4539b1f0d0876a53be15